sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusMessage.cs

```diff
@@ -108,6 +108,26 @@ public ServiceBusMessage(ServiceBusReceivedMessage receivedMessage)
         /// </remarks>
         public BinaryData Body { get; set; }
 
+        internal enum AmqpBodyType
```

Comment:
I think we're mixing concepts here.  The details of the transport protocol should be abstracted from the message and pluggable.   The message should only know that there "some kind of transport stuff that I need to hold" but none of the details.

Any manipulation of the transport stuff should come from the extensions for that specific protocol.   Otherwise, we potentially end up with a very fat model that tries to encapsulate details for every protocol that can be used.

Reply:
I agree. I did this as a first blush to see how everything would work around. I'll move this to some domain-specific area (meaning, AMQP specific). I'll figure out a way to have ServiceBusMessage not know these other interpretations of the information exists.

Reply:
(copy pasta from above): I created `AmqpTransportBody`. This implemented `ITransportBody`. A `ServiceBusMessage` contains an internal `TransportBody`. The `Body` property on `ServiceBusMessage` simply exposes the `Body` property of the `TransportBody` via the `ITransportBody` interface.